examples/e2e-test-dataflow/src/test/resources/e2etest/dataflow-e2e-test.feature

```diff
@@ -0,0 +1,49 @@
+Feature: Dataflow Integration testing
+  End to End testing for the Dataflow GCS to BQ
+
+  Background:
+  Scenario: Download the jar to test
+    Given A jar at URL "gs://<<GCS_SOURCE_BUCKET>>/snapshot/com/demo/bikeshare-csv-to-bq/1.0-SNAPSHOT/bikeshare-csv-to-bq-1.0-20200116.200619-1.jar"
+    Then Download the jar to "data/code/" as "dataflow.jar"
+    And Check the jar is downloaded successfully
+
+  @IngestionE2ESmokeTest
+  Scenario: Data is ingested successfully
+    Given Using the "dataflow.jar" and the following properties
```

Comment:
Ideally a scenario should be arranged as Given (defining the required state), When (Performing an action), Then (defining the expected behavior).  In case if each of these clauses requires multiple steps then these steps can be combined using AND clause.

For e.g: In the IngestionE2ESmokeTest scenario can be changed as:
**Given** Using the "dataflow.jar" and the following properties
      | project | <<PROJECT_ID>> |
      | serviceAccount | dataflow-runner-gcs-to-bq@<<PROJECT_ID>>.iam.gserviceaccount.com  |
      | dataset        | test_e                                           |
      | tempLocation     | gs://<<TEMP_GCS_BUCKET>>/temp |
      | runner           | Dataflowrunner |
      | inputFilePattern  | gs://<<INPUT_GCS_BUCKET>>/*.csv |
      | tableName         | testTable                 |
      | errorTableName    | testTableErr               |
    AND tempLocation exists
    And input File Pattern exists
    And output Bigquery dataset exists
    **WHEN** Run dataflow application main class being "com.demo.dataflow.GoBikeToBigQuery"
     AND Wait for the Dataflow application to complete, maximum wait time 10 minutes
    **THEN**  Check The query "SELECT count(*) as count FROM %s.%s" returns "21" records
      AND  The Query "SELECT count(*) as count FROM %s.%s" returns "5" records
      AND scenario resources were cleaned successfully   (Can be skipped as well since these were handled already in step definition using @After scenario hook)

Reply:
Thanks @prathapreddy123 I have made the changes.